File: TestLeetCode/Stack/DailyTemperatures.cs
public class DailyTemperaturesSolution
{
    public int[] DailyTemperatures(int[] temperatures)
    {
        int[] res = new int[temperatures.Length];

        // This is a good example of new C# feature without using a POCO class.
        // (int temp, int index) is equivalent to the new class type
        Stack<(int temp, int index)> stack = new Stack<(int, int)>();

        for (int i = 0; i < temperatures.Length; i++)
        {
            int t = temperatures[i];
            while (stack.Count > 0 && t > stack.Peek().temp)
            {
                var (_, index) = stack.Pop();
                res[index] = i - index;
            }
            // notice that here, we need double ()
            stack.Push((t, i));
        }
        return res;
    }
}

File: python/stack/daily_temperatures.py
class DailyTemperatures:
    def daily_temperatures(self, temperatures: list[int]) -> list[int]:
        result, stack = [0] * len(temperatures), []
        for index, temperature in enumerate(temperatures):
            while stack and temperatures[stack[-1]] < temperature:
                previous = stack.pop(); result[previous] = index - previous
            stack.append(index)
        return result
